demo-server/src/main/java/com/illia/server/request_processor/RequestProcessorImpl.java

```diff
@@ -27,24 +29,29 @@ public ResponseEntity<Object> proceedDownloadFile(String fileName) {
                 return ResponseEntity.badRequest().body("No such file!".getBytes());
             }
         } catch (Exception ex) {
-            return ResponseEntity.internalServerError().build();
+            var errorMsg = "FileHolder does not respond!";
+            log.error(errorMsg, ex);
+            return ResponseEntity.internalServerError().body(errorMsg);
```

Comment:
"FileHolder does not respond" means nothing to user. Internal error would be enough.